roles/adv-prog-pkgs/vars/main.yml

```diff
@@ -1,2 +1,24 @@
 ---
-# vars file for adv-prog-pkgs
\ No newline at end of file
+# vars file for adv-prog-pkgs
+adv_prog_pkgs_adv_lang:
+  - build-essential
+  - g++
+  - gcc
+  - gdb
+  - logisim
+  - valgrind
+
+adv_prog_pkgs_souce_control:
+  - git
+  - gitg
+  - mercurial
+  - meld
+
+adv_prog_pkgs_code_editor:
+  - astyle
+  - bvi
+  - check
+  - gedit
+  - indent
+  - nano
+  
```

Comment:
Hate to be nitpicky, but can you remove the whitespace on this line? Otherwise there isn't actually a newline at the end of the file (`'\n'`) it's a space (`' '`).

```suggestion

```